File: benchmarks/vm/07_string_scan.py
import time

line = "SET:12345:payload"
t0 = time.monotonic()
total = 0
for _ in range(50000):
    if line.startswith("SET:") and "payload" in line:
        total += len(line[4:9])
total_ms = (time.monotonic() - t0) * 1000
print(f"total = {total}")
print(f"BENCH_RESULT: {total_ms:.3f} ms")

File: benchmarks/vm/07_string_scan.rb
line = "SET:12345:payload"
t0 = Process.clock_gettime(Process::CLOCK_MONOTONIC)
total = 0
50_000.times do
  if line.start_with?("SET:") && line.include?("payload")
    total += line[4, 5].length
  end
end
total_ms = (Process.clock_gettime(Process::CLOCK_MONOTONIC) - t0) * 1000
puts "total = #{total}"
puts format("BENCH_RESULT: %.3f ms", total_ms)
